core/templates/dev/head/components/question-directives/questions-list/questions-list.directive.html

```diff
@@ -26,7 +26,7 @@
       ng-click="$ctrl.editQuestion(question.summary)">
     <td> <[$ctrl.getQuestionIndex($index)]> </td>
     <td> <[$ctrl.truncatedQuestionSummaries[$index]]> </td>
-    <td ng-if="!$ctrl.getSkill()">{{question.skill_description}}</td>
+    <td ng-if="!$ctrl.getSkill()"> <[$ctrl.getSkillDescription(question.skill_descriptions)]> </td>
```

Comment:
Odd that you're using underscore convention here. Shouldn't question be an instance of a _frontend_ domain object with a skillDescriptions property? We should be passing around those domain objects, not backend dicts.

Reply:
I think the keys of the dict was never changed after returning from the backend.

Reply:
It should be a javascript dict parsed from JSON right?

Reply:
No it should be a frontend domain object (i.e. created using QuestionObjectFactory). Can you check that it is?

Reply:
<img width="300" alt="Screen Shot 2019-06-17 at 8 49 15 PM" src="https://user-images.githubusercontent.com/32754302/59652198-8adaaf00-9141-11e9-8a05-6cdb875bcdd2.png">
<img width="564" alt="Screen Shot 2019-06-17 at 8 49 32 PM" src="https://user-images.githubusercontent.com/32754302/59652199-8ca47280-9141-11e9-8086-edbe0c3bd76e.png">
I printed it out, and seems like it has the same structure as the return dict from the backend.

Reply:
You should actually be looking into the frontend typescript code to see where "questions" comes from. I just did that and it seems to come from getQuestionSummaries() (which actually suggests that "question" is misnamed in the template, it should be "questionSummary") and also it seems to be just the backend dict without any conversion. We should actually convert the backend dict to a domain object (see the files in domain/question/ for examples) and pass the domain object around instead.

If you could fix the "questionSummary" naming, that'd be great! It's fine to do the domain object in a separate PR instead, but please file an issue to keep track if you decide to do that.

Reply:
Yeah I agree it's better to name it as questionSummary, however, I don't think this dict could be converted to a questionSummary domain object. The 'summary' part in the return_dict could be, but the 'skill_descriptions' part is actually outside of the domain object.

Reply:
Oh, that's exactly what I wanted to avoid and that's actually the reason we use domain objects -- so that we do not just add properties to a dict midway. We really ought to fix this soonish.

There should be a well-defined QuestionSummary domain object that does _not_ have the skill description. There should also be a service that stores and maintains a skill-id-to-description mapping. Use the service to convert skill ID to description when needed.

Reply:
Ok I will file an issue to keep track of this.

Reply:
OK thanks, please link to the issue here once you've done so.